src/main/java/org/wahlzeit_revisited/service/PhotoService.java

```diff
@@ -0,0 +1,122 @@
+package org.wahlzeit_revisited.service;
+
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.Response;
+import org.wahlzeit_revisited.dto.PhotoDto;
+import org.wahlzeit_revisited.model.Photo;
+import org.wahlzeit_revisited.model.PhotoFactory;
+import org.wahlzeit_revisited.model.User;
+import org.wahlzeit_revisited.repository.PhotoRepository;
+import org.wahlzeit_revisited.utils.SysLog;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
+public class PhotoService {
+
+    @Inject
+    public Transformer transformer;
+    @Inject
+    public PhotoRepository repository;
+    @Inject
+    public PhotoFactory factory;
+
+    /*
+     * business methods
+     */
+
+    public Response getPhotos() throws SQLException {
+        List<Photo> photoList = repository.findAll();
+
+        List<PhotoDto> responseDto = new ArrayList<>(photoList.size());
+        for (Photo photo : photoList) {
+            responseDto.add(transformer.transform(photo));
+        }
+        return Response.ok(responseDto).build();
+    }
```

Comment:
Why do we even see DTOs here? I thought the JEE framework hides this. Also, if you have to do it by hand, converting an object into a DTO is the perfect Serializer task (which what exists for) and it is the same code as for saving to the SQL database.

Reply:
Java EE hides the the serialization from POJO to JSON. The purpose of the DTOs are here for a clear API between the frontend and backend

Due it's possible to send the Entity-class, and hide some unnecessary fields with some annotations, like one would see in mircroservice architectures like the [ECB Pattern](https://github.com/LPCISIIE/REST-JPA)
I choose this variant, especially with the transformer, as I've applied it the patch-crawler framework in my bachelor thesis and all productive applications at work